server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -0,0 +1,279 @@
+        return this.setting;
+    }
+
+    /**
+     * Gets the generic type of the wrapped setting.
+     *
+     * @return The wrapped setting's generic type.
+     */
+    public SettingType getType() {
+        return this.type;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Setting<?> createSetting(
+        SettingType type,
+        String key,
+        Object defaultValue,
+        WriteableSetting fallback,
+        Property[] propertyArray
+    ) {
+        switch (type) {
+            case Boolean:
+                return fallback == null
+                    ? Setting.boolSetting(key, (boolean) defaultValue, propertyArray)
+                    : Setting.boolSetting(key, (Setting<Boolean>) fallback.getSetting(), propertyArray);
+            case Integer:
+                return fallback == null
+                    ? Setting.intSetting(key, (int) defaultValue, propertyArray)
+                    : Setting.intSetting(key, (Setting<Integer>) fallback.getSetting(), propertyArray);
+            case Long:
+                return fallback == null
+                    ? Setting.longSetting(key, (long) defaultValue, propertyArray)
+                    : Setting.longSetting(key, (Setting<Long>) fallback.getSetting(), propertyArray);
+            case Float:
+                return fallback == null
+                    ? Setting.floatSetting(key, (float) defaultValue, propertyArray)
+                    : Setting.floatSetting(key, (Setting<Float>) fallback.getSetting(), propertyArray);
+            case Double:
+                return fallback == null
+                    ? Setting.doubleSetting(key, (double) defaultValue, propertyArray)
+                    : Setting.doubleSetting(key, (Setting<Double>) fallback.getSetting(), propertyArray);
+            case String:
+                return fallback == null
+                    ? Setting.simpleString(key, (String) defaultValue, propertyArray)
+                    : Setting.simpleString(key, (Setting<String>) fallback.getSetting(), propertyArray);
+            case TimeValue:
+                return fallback == null
+                    ? Setting.timeSetting(key, (TimeValue) defaultValue, propertyArray)
+                    : Setting.timeSetting(key, (Setting<TimeValue>) fallback.getSetting(), propertyArray);
+            case ByteSizeValue:
+                return fallback == null
+                    ? Setting.byteSizeSetting(key, (ByteSizeValue) defaultValue, propertyArray)
+                    : Setting.byteSizeSetting(key, (Setting<ByteSizeValue>) fallback.getSetting(), propertyArray);
+            case Version:
+                // No fallback option on this method
+                return Setting.versionSetting(key, (Version) defaultValue, propertyArray);
+            default:
+                // This Should Never Happen (TM)
+                throw new UnsupportedOperationException("A SettingType has been added to the enum and not handled here.");
```

Comment:
I think it is better to use `IllegalStateException` instead of `UnsupportedOperationException`: the operation is in fact supported but the type of the setting is unexpected.

Reply:
Resolved but not changed?

Reply:
I thought I had changed it - must have been overwritten somehow